sklearn/impute.py

```diff
@@ -1207,6 +1207,14 @@ def fit(self, X, y=None):
             raise ValueError("'features' has to be either 'missing-only' or "
                              "'all'. Got {} instead.".format(self.features))
 
+        if sparse.issparse(X):
+            # missing_values = 0 not allowed with sparse data as it would
+            # force densification
+            if self.missing_values == 0:
+                raise ValueError("Imputation not possible when missing_values "
+                                 "== 0 and input is sparse. Provide a dense "
```

Comment:
It's not Imputation here. Maybe something like "Sparse input with missing_values=0 is not supported."

Reply:
Ok, I changed it.